### PR TITLE
use yml settings and dor_services initializer for configuration

### PR DESCRIPTION
**Pull request by ndushay**
The impetus for this PR is #78 -- have job outputs in an appropriate location.   This should be a configuration setting, and has nothing to do with dor-services.

It seemed a propitious time address #169 to move config data into config/settings/#{RAILS_ENV}.yml file and having the dor_services initializer turn the settings into ruby code.

Connects to #169 which will be complete when config/cli_environments is removed from shared_configs and from preassembly code base.

Connects to #78 which will use the `job_output_parent_dir` setting.

Note that the edit to reporting_old class helped me test this on the console on -stage;  I know the class is going away.

### CODE REVIEW

File: app/lib/pre_assembly/reporting_old.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # encoding: UTF-8
 
 module PreAssembly
-  module Reporting
+  module ReportingOld
```
Comment (atz) [inline app/lib/pre_assembly/reporting_old.rb]: I think this change unintentionally crept in

Comment (ndushay) [reply app/lib/pre_assembly/reporting_old.rb]: actually, you reviewed the PR before I hit save on my edits to comment above. 

>Note that the edit to reporting_old class helped me test this on the console on -stage; I know the class is going away.

essentially, without this stupid edit, it has two modules called the same thing so I couldn't execute the one I wanted when manually testing that I had all the settings correct and accounted for.

Comment (atz) [reply app/lib/pre_assembly/reporting_old.rb]: Ah, that makes sense.  I forgot you already copied that class.

**Comment by ndushay** [comment]
@atz made the suggestion that I create the job output directories referenced in the settings.yml files.  This is problematic because `.gitignore` contains `log/*`.   

I could make the default directory "log" and not have settings override for dev and test environments ... would that be the way to go here?

**Comment by atz** [comment]
Ah, no need, we can deal w/ things later.

**Pull request merged**